pyext/src/macros.py

```diff
@@ -552,7 +573,18 @@ def execute_macro(self):
             if save_frame:
                 print("--- frame %s score %s " % (str(i), str(score)))
 
-                if not self.test_mode:
+                if self.nest and not math.isnan(score):
+                    likelihood_for_sample = 1
+                    for rstrnt in self.nestor_restraints:
+                        likelihood_for_sample = (
+                            likelihood_for_sample * rstrnt.get_likelihood()
+                        )
+                    sampled_likelihoods.append(likelihood_for_sample)
+
+                if self.nest and math.isnan(score):
+                    sampled_likelihoods.append(math.nan)
```

Comment:
It's a bit hard to read this way and a little inefficient (since you check the conditions twice). Consider instead

    if self.nest:
        if math.isnan(score):
            sampled_likelihoods.append(math.nan)
        else:
            sampled_likelihoods.append(<product of likelihood over all restraints>)

Reply:
Yeah, I guess I missed it. This is fixed (Updated lines 575-582)